test/NuGet.Core.Tests/NuGet.Configuration.Test/SearchTreeTests.cs

```diff
@@ -0,0 +1,344 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Linq;
+using FluentAssertions;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace NuGet.Configuration.Test
+{
+    public class SearchTreeTests
+    {
+        [Fact]
+        public void SearchTree_WithOneSource()
+        {
+            // Arrange
+            using var mockBaseDirectory = TestDirectory.Create();
+            var configPath1 = Path.Combine(mockBaseDirectory, "NuGet.Config");
+            SettingsTestUtils.CreateConfigurationFile(configPath1, @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageNamespaces>
+        <clear />
+        <packageSource key=""nuget.org"">
+            <namespace id=""stuff"" />
+        </packageSource>
+    </packageNamespaces>
+</configuration>");
+            var settings = Settings.LoadSettingsGivenConfigPaths(new string[] { configPath1 });
+
+            // Act & Assert
+            var configuration = PackageNamespacesConfiguration.GetPackageNamespacesConfiguration(settings);
+            Assert.True(configuration.IsNamespacesEnabled);
+
+            configuration.Namespaces.Should().HaveCount(1);
+            var packageSourcesMatchFull = configuration.GetConfiguredPackageSources("stuff");
+            Assert.Equal(1, packageSourcesMatchFull.Count);
+            Assert.Equal("nuget.org", packageSourcesMatchFull.First());
+
+            var packageSourcesMatchPartial = configuration.GetConfiguredPackageSources("stu");
+            Assert.Null(packageSourcesMatchPartial);
+
+            var packageSourcesNoMatch = configuration.GetConfiguredPackageSources("random");
+            Assert.Null(packageSourcesNoMatch);
+        }
+
+        [Fact]
+        public void SearchTree_WithOneSourceMultipart()
```

Comment:
this is really about globbing right?

What is multipart supposed to represent here?

Reply:
Removed.